### PR TITLE
phpExtensions.mongodb: 1.15.0 -> 1.16.1

### PR DESCRIPTION
**Pull request by drupol**
This PR is on hold because we are unable to build the extension for `x86_64-darwin` architecture.
An issue has been opened upstream at https://jira.mongodb.org/browse/PHPC-2257

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb php82Extensions.mongodb php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb php82Extensions.mongodb php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb php82Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php82Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb
@GrahamcOfBorg build php82Extensions.mongodb
@GrahamcOfBorg build php83Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

File: pkgs/top-level/php-packages.nix [hunk 1]
```diff
@@ -244,7 +244,10 @@ lib.makeScope pkgs.newScope (self: with self; {
 
     memcached = callPackage ../development/php-packages/memcached { };
 
-    mongodb = callPackage ../development/php-packages/mongodb { };
+    mongodb = pkgs.darwin.apple_sdk_11_0.callPackage ../development/php-packages/mongodb {
+      inherit (pkgs.darwin.apple_sdk_11_0.frameworks) Security;
+      inherit (pkgs.darwin) CommonCrypto;
```
Comment (wegank) [inline pkgs/top-level/php-packages.nix:249]:
```suggestion
      inherit (pkgs.darwin.apple_sdk_11_0) Libsystem;
```

File: pkgs/development/php-packages/mongodb/default.nix [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
 , pkg-config
 , cyrus_sasl
 , icu64
 , openssl
 , snappy
 , zlib
-, darwin
-}:
+, Security
+, CommonCrypto
```
Comment (wegank) [inline pkgs/development/php-packages/mongodb/default.nix:14]:
```suggestion
, Libsystem
```

File: pkgs/development/php-packages/mongodb/default.nix [hunk 3]
```diff
@@ -25,12 +35,16 @@ buildPecl {
     snappy
     zlib
     pcre2
-  ] ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security;
+  ] ++ lib.optionals stdenv.isDarwin [
+    Security
+    CommonCrypto
```
Comment (wegank) [inline pkgs/development/php-packages/mongodb/default.nix:40]:
```suggestion
    Libsystem
```

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.mongodb

**Pull request merged**